delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/BlazegraphViews.scala

```diff
@@ -1,17 +1,313 @@
 import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.model._
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.RemoteContextResolution
+import ch.epfl.bluebrain.nexus.delta.sdk.cache.{KeyValueStore, KeyValueStoreConfig}
+import ch.epfl.bluebrain.nexus.delta.sdk.eventlog.EventLogUtils
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.ExpandIri
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.JsonLdSourceProcessor.JsonLdSourceDecoder
+import ch.epfl.bluebrain.nexus.delta.sdk.model.IdSegment.IriSegment
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
 import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.Permission
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.{Project, ProjectRef}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.Pagination.FromPagination
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.ResultEntry.UnscoredResultEntry
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchResults.UnscoredSearchResults
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Envelope, IdSegment, Label, TagLabel}
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
 import ch.epfl.bluebrain.nexus.delta.sdk.utils.{IOUtils, UUIDF}
-import monix.bio.{IO, UIO}
+import ch.epfl.bluebrain.nexus.delta.sdk.{Organizations, Permissions, Projects}
+import ch.epfl.bluebrain.nexus.sourcing.SnapshotStrategy.NoSnapshot
+import ch.epfl.bluebrain.nexus.sourcing.processor.EventSourceProcessor.persistenceId
+import ch.epfl.bluebrain.nexus.sourcing.processor.ShardedAggregate
+import ch.epfl.bluebrain.nexus.sourcing.projections.StreamSupervisor
+import ch.epfl.bluebrain.nexus.sourcing.{Aggregate, EventLog, PersistentEventDefinition}
+import com.typesafe.scalalogging.Logger
+import fs2.Stream
+import io.circe.Json
+import monix.bio.{IO, Task, UIO}
+import monix.execution.Scheduler
+
+/**
+  * Operations for handling Blazegraph views.
+  */
+final class BlazegraphViews(
+    agg: BlazegraphViewsAggregate,
+    eventLog: EventLog[Envelope[BlazegraphViewEvent]],
+    index: BlazegraphViewsCache,
+    projects: Projects,
+    orgs: Organizations,
+    sourceDecoder: JsonLdSourceDecoder[BlazegraphViewRejection, BlazegraphViewValue]
+)(implicit rcr: RemoteContextResolution) {
+
+  /**
+    * Create a new Blazegraph view where the id is either present on the payload or self generated.
+    *
+    * @param project  the project of to which the view belongs
+    * @param source   the payload to create the view
+    */
+  def create(project: ProjectRef, source: Json)(implicit
+      subject: Subject
+  ): IO[BlazegraphViewRejection, BlazegraphViewResource] = {
+    for {
+      p                <- projects.fetchActiveProject(project)
+      (iri, viewValue) <- sourceDecoder(p, source)
+      res              <- eval(CreateBlazegraphView(iri, project, viewValue, source, subject), p)
+    } yield res
+  }.named("createView", moduleType)
```

Comment:
Maybe `createBlazegraphView` instead? It would be useful to have different spans for different views, since from the API perspective we would not always be able to differentiate them